### ISSUE TITLE
Latest Visual Studio 2022 update v17.5.1 breaks docfx

### ISSUE THREAD

**Issue by andyb70** [dotnet/docfx#8465]
**Describe the bug**
The latest Visual Studio update to v17.5.1 bumps the .NET SDK to version 7.0.201 and this appears to have caused dependency issues when running the `docfx docfx_project/docfx.json --serve` command.

I recieve the following failures against multiple .csproj files
````
[Failure] Msbuild failed when processing the file 'C:\SourceCode\Manatee\Version2\Caboodal.Manatee.Logic.Widget\Caboodal.Manatee.Logic.Widget.csproj' with message: C:\Program Files\dotnet\sdk\7.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets: (90, 5): The "ProcessFrameworkReferences" task failed unexpectedly.
System.IO.FileLoadException: Could not load file or assembly 'NuGet.Frameworks, Version=6.5.0.136, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. Could not find or load a specific file. (0x80131621)
````

I subsequently receive lots of build errors for example

````
C:\SourceCode\Manatee\Version2\Caboodal.Manatee.DataTransferObjects\Authentication\DtoLoginResult.cs(1,7): error CS0246: The type or namespace name 'Newtonsoft' could not be found (are you missing a using directive or an assembly reference?)
C:\SourceCode\Manatee\Version2\Caboodal.Manatee.DataTransferObjects\Authentication\DtoPasswordReset.cs(1,7): error CS0246: The type or namespace name 'System' could not be found (are you missing a using directive or an assembly reference?)
C:\SourceCode\Manatee\Version2\Caboodal.Manatee.DataTransferObjects\Authentication\DtoPasswordReset.cs(2,7): error CS0246: The type or namespace name 'Newtonsoft' could not be found (are you missing a using directive or an assembly reference?)
C:\SourceCode\Manatee\Version2\Caboodal.Manatee.DataTransferObjects\Checklists\DtoActivityHistory.cs(1,7): error CS0246: The type or namespace name 'System' could not be found (are you missing a using directive or an assembly reference?)
C:\SourceCode\Manatee\Version2\Caboodal.Manatee.DataTransferObjects\Checklists\DtoActivityHistory.cs(2,7): error CS0246: The type or namespace name 'Newtonsoft' could not be found (are you missing a using directive or an assembly reference?)
````

**To Reproduce**
Update Visual Studio 2022 to v17.5.1 and run a docfx project that targets .NET6.0 class library project.

**Context (please complete the following information):**
- OS: Windows 11 Professional
- Docfx version: v.2.62.1

**Comment by andyb70** [comment]
After spending a couple of days looking at this with the a copy of the repo I have also noticed that the static class `DotnetApiCatalog` has a constructor that calls...

````
MSBuildLocator.RegisterDefaults()
````

This throws the following error:

````
System.TypeInitializationException: The type initializer for 'Microsoft.DocAsCode.Dotnet.DotnetApiCatalog' threw an exception.
 ---> System.InvalidOperationException: No instances of MSBuild could be detected.
Try calling RegisterInstance or RegisterMSBuildPath to manually register one.
   at Microsoft.Build.Locator.MSBuildLocator.RegisterDefaults()
````

I thought this odd so I created a .NET Framework 4.8 console app that calls this method and it works however, I created .NET 6.0 console app and I get the same error message as above.

I have subsequently found this issue [#197](https://github.com/microsoft/MSBuildLocator/issues/197) on the Microsoft.Build.Locator repo.

Whether there is any link I don't know, and why the error message is different I'm also unsure but I thought I would mention it.

**Comment by yufeih** [comment]
What did you get from `dotnet --list-sdks` command? `Microsoft.Build.Locator` in a .NET Core app requires .NET Core SDK installed on the system.

**Comment by andyb70** [comment]
````
λ dotnet --list-sdks
7.0.201 [C:\Program Files\dotnet\sdk]
````

**Comment by andyb70** [comment]
My colleague had his development environment still configured to SDK 7.0.103 and his worked we then upgraded his dev environment to SDK 7.0.201 and his also failed with the same error.

**Comment by mawosoft** [comment]
The problem most likely is that the docfx package already comes with NuGet.\*/`6.4.0` assemblies included. That version matches the NuGet.\* assemblies included in the dotnet SDK prior to `7.0.200`. From there on, the SDK uses NuGet.*/`6.5.0`, which the msbuild tasks then cannot load into the docfx process.

At least that would be my guess.

**Comment by andyb70** [comment]
Thank you @yufeih everything is now working!
